components/webhook-mgt/org.wso2.carbon.identity.webhook.management/src/main/java/org/wso2/carbon/identity/webhook/management/internal/service/impl/WebhookManagementServiceImpl.java

```diff
@@ -0,0 +1,172 @@
+import org.wso2.carbon.identity.webhook.management.api.exception.WebhookMgtException;
+import org.wso2.carbon.identity.webhook.management.api.model.Webhook;
+import org.wso2.carbon.identity.webhook.management.api.model.WebhookDTO;
+import org.wso2.carbon.identity.webhook.management.api.model.WebhookStatus;
+import org.wso2.carbon.identity.webhook.management.api.service.WebhookManagementService;
+import org.wso2.carbon.identity.webhook.management.internal.dao.WebhookManagementDAO;
+import org.wso2.carbon.identity.webhook.management.internal.dao.impl.CacheBackedWebhookManagementDAO;
+import org.wso2.carbon.identity.webhook.management.internal.dao.impl.WebhookManagementDAOFacade;
+import org.wso2.carbon.identity.webhook.management.internal.dao.impl.WebhookManagementDAOImpl;
+import org.wso2.carbon.identity.webhook.management.internal.util.WebhookManagementExceptionHandler;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+/**
+ * Implementation of WebhookManagementService.
+ * This class uses WebhookManagementFacade to handle webhook operations.
+ * TODO: Check the supported event and schema using Meta data service
+ */
+public class WebhookManagementServiceImpl implements WebhookManagementService {
+
+    private static final Log LOG = LogFactory.getLog(WebhookManagementServiceImpl.class);
+    private static final WebhookManagementDAO DAO_FACADE =
+            new WebhookManagementDAOFacade(new CacheBackedWebhookManagementDAO(new WebhookManagementDAOImpl()));
+
+    @Override
+    public WebhookDTO createWebhook(Webhook webhook, String tenantDomain) throws WebhookMgtException {
+
+        if (LOG.isDebugEnabled()) {
+            LOG.debug(String.format("Creating webhook with endpoint: %s for tenant: %s",
+                    webhook.getEndpoint(), tenantDomain));
+        }
+        int tenantId = IdentityTenantUtil.getTenantId(tenantDomain);
+        if (DAO_FACADE.isWebhookEndpointExists(webhook.getEndpoint(), tenantId)) {
+            throw WebhookManagementExceptionHandler.handleClientException(
+                    ErrorMessage.ERROR_CODE_WEBHOOK_ENDPOINT_ALREADY_EXISTS, webhook.getEndpoint());
+        }
+        String generatedWebhookId = UUID.randomUUID().toString();
+        webhook.setId(generatedWebhookId);
+
+        // Set default status to INACTIVE if not provided
+        if (webhook.getStatus() == null) {
+            webhook.setStatus(WebhookStatus.INACTIVE);
+        }
+
+        DAO_FACADE.createWebhook(webhook, tenantId);
+        Webhook createdWebhook = DAO_FACADE.getWebhook(webhook.getId(), tenantId);
+        return WebhookDTO.Builder.fromWebhook(createdWebhook).build();
+    }
+
+    @Override
+    public WebhookDTO getWebhook(String webhookId, String tenantDomain) throws WebhookMgtException {
+
+        if (LOG.isDebugEnabled()) {
+            LOG.debug(String.format("Retrieving webhook with ID: %s for tenant: %s",
+                    webhookId, tenantDomain));
+        }
```

Comment:
If these debug logs gets added in the service level let's not repeat in facade or dao level